scripts/bdc/get_bdc_studies_from_gen3.py

```diff
@@ -64,305 +69,500 @@ def download_gen3_list(input_url, download_limit=GEN3_DOWNLOAD_LIMIT):
-                    if set(enums) != set(encs.keys()):
-                        logging.error(f"`constraints.enum` ({row['constraints.enum']}) and `encodings` ({row['encodings']}) do not match.")
-                        counters['enum_encoding_mismatch'] += 1
-
-                # Variable type.
-                typ = row.get('type')
-                if encs:
-                    typ = 'encoded value'
-                if typ:
-                    type_element = ETree.SubElement(variable, 'type')
-                    type_element.text = typ
-
-                # We currently ignore metadata fields not usually filled in for input VLMD files:
-                # ordered, missingValues, trueValues, falseValues, repo_link
-
-                # We currently ignore all standardMappings: standardsMappings.type, standardsMappings.label,
-                # standardsMappings.url, standardsMappings.source, standardsMappings.id
-                # We currently ignore all relatedConcepts: relatedConcepts.type, relatedConcepts.label,
-                # relatedConcepts.url, relatedConcepts.source, relatedConcepts.id
-
-                # We currently ignore all univarStats vars: univarStats.median, univarStats.mean, univarStats.std,
-                # univarStats.min, univarStats.max, univarStats.mode, univarStats.count,
-                # univarStats.twentyFifthPercentile, univarStats.seventyFifthPercentile,
-                # univarStats.categoricalMarginals.name, univarStats.categoricalMarginals.count
-        else:
-            # This shouldn't be needed, since Click should catch any file format not in the accepted list.
-            raise RuntimeError(f"Unsupported file format {file_format}")
-
-        # Write out dbGaP XML.
-        xml_str = ETree.tostring(data_table, encoding='unicode')
-        pretty_xml_str = minidom.parseString(xml_str).toprettyxml()
-        print(pretty_xml_str, file=output)
-
-        # Display counters.
-        logging.info(f"Counters: {json.dumps(counters, sort_keys=True, indent=2)}")
+    :param output: The CSV file to be generated.
+    :param bdc_gen3_base_url: The BDC Gen3 base URL (i.e. everything before the
+    `/mds/...`). Defaults to https://gen3.biodatacatalyst.nhlbi.nih.gov/.
+    """
+    study_info_list = retrieve_study_info_list(bdc_gen3_base_url)
+
+    write_list_to_csv_file(study_info_list, output)
+
+
+    if kgx_file:
+        (nodes, edges) = make_kgx_lists(study_info_list)
+        logging.info("Writing out %d kgx nodes to file %s", len(nodes),
+                      kgx_file)
+        json.dump(make_kgx(nodes, edges), kgx_file, indent=2)
+
+# def _depricated_code():
+#     """This code previously came after the csv_writer.writerow column but
+#     also came after an exit(0) statement that made it unreachable. It has been
+#     wrapped in a function definition to help code organization.
+
+#     Possibly consider just deleting this whole block?
```

Comment:
Yeah, let's just delete this section: it looks like it's duplicated from code to generate a dbGaP XML file, which we definitely don't need to do here. Sorry for not deleting it earlier!